### ISSUE TITLE
Change medication ordering to use formulations

### ISSUE THREAD

**Issue by christillman** [christillman/encounterpro_os#20]
We can use the patient 'Marvin Price'.

For illustrative purposes, I will be using the legacy application since the patient data remains the same.

From the chart tab, search for the patient : last name-'Price' and first name 'Marvin'.

![unnamed](https://github.com/christillman/encounterpro_os/assets/25717203/b2402499-2e7a-4322-879e-2c444f66c775)

Select the patient name bar. This will open the patient demographics page. On  this page, select the 'View Chart' button from the menu located on the bottom of the page.

![unnamed](https://github.com/christillman/encounterpro_os/assets/25717203/de3bf291-4100-4390-93d8-b7fdb379ee10)

Selecting this button will open the patient's chart page.  
![unnamed](https://github.com/christillman/encounterpro_os/assets/25717203/f3e468ac-0b33-4c9d-bf0e-c8ac3a1a96f7)

On this chart page, select the 'capsule' icon below the 'treatments with no assssment' title.

![unnamed](https://github.com/christillman/encounterpro_os/assets/25717203/193058b4-3752-480f-af67-f1969240ad9c)

It will display a menu containing buttons. From this menu, select the 'Edit Prescription' button. This button is linked to the 'Edit Prescription' service and wherever it is configured in the application, it will open the 'edit prescription' page. See below:

![unnamed](https://github.com/christillman/encounterpro_os/assets/25717203/39046fbe-cf7d-4a09-b6db-553b6f8f73ee)

Could this page be replaced with our current updated medication ordering page? I think this could be the same page that I had put in a request to suppress the 'dose based on' field.

The difference is, the button that needs to be selected to open the same page is the 'Add Medication' button and it is located on the 'Current Medications' page.

To access the 'Current Medications' page:
From the patient chart, select the white box labeled 'Meds' located on the top mid section of the patient chart page. See below

![unnamed](https://github.com/christillman/encounterpro_os/assets/25717203/eb329617-202b-4f8d-91d7-2bb3d8ad5ac5)

It will show the 'Current Medications' page seen below.

![unnamed](https://github.com/christillman/encounterpro_os/assets/25717203/463fd7b0-81b5-49be-937f-03c59b768823)

The 'Add Meds' button is located on the right side of the page. See below:
![unnamed](https://github.com/christillman/encounterpro_os/assets/25717203/6dd72961-dc31-419f-b154-ff0273dcd13b)

Selecting the 'Add Meds' page opens the 'Select Prescription(s) page

![unnamed](https://github.com/christillman/encounterpro_os/assets/25717203/d9b4c8b8-eb1e-4023-b888-ae5ea7c22d7e)

Selecting the prescription of choice (I chose Allegra), opens the 'Current Medications' page

![unnamed](https://github.com/christillman/encounterpro_os/assets/25717203/eec52da2-3ef6-4de2-9eaf-f5c8306f6939)

Could this page be replaced by our current drug formulation page? This will eliminate the need to suppress the 'dose based on' field.

Therefore the process would be: Select the 'Add Meds' field

![unnamed](https://github.com/christillman/encounterpro_os/assets/25717203/b5061a88-cee2-419e-a6bb-a5c970ef9214)

Select the prescription of choice from the page that shows next seen below:
![unnamed](https://github.com/christillman/encounterpro_os/assets/25717203/9b0d2254-2336-4470-a9b9-cc77058788ea)

Selecting the medication of choice should open our new drug formulation page

Followed lastly by our revised current drug ordering page.

**Comment by christillman** [comment]
So, summarizing:

When the user selects either "Edit Prescription" button or "Add Meds" button (pictures & process above), the user should then follow the process same we worked on earlier when they select 'Any Trtmt"

![image](https://github.com/christillman/encounterpro_os/assets/25717203/81e7e7da-eb79-4655-bc16-8761aabf71da)

followed by "Prescription Drugs"

![image](https://github.com/christillman/encounterpro_os/assets/25717203/650201e1-926f-4257-bc64-860566cda69f)

ie select a drug

![image](https://github.com/christillman/encounterpro_os/assets/25717203/25f1b269-7748-46c2-996a-d34eb0b04c7c)

followed by select a formulation
![image](https://github.com/christillman/encounterpro_os/assets/25717203/49ee9586-a6a2-4952-b5f4-5b37b5a3568e)

resulting in Selected Item(s)

**Comment by christillman** [comment]
Here is what's happening when I select the Capsule icon. For Marvin's existing Prelone prescription, I get "No formulation selected" because it was created before our new code was added.

```
// Make sure we have a formulation
if isnull(treatment.form_rxcui) then
	log.log(this, "u_component_service_get_medication.xx_do_service:0025", "No formulation selected", 4)
	return 2
end if
```

At this point, if no formulation has been selected, should the user be directed to the formulation selection page? Then they could continue to the next section, which is where I get to if I add a prescription now (with formulation selected).

```
// If it's a new treatment then use the dosing screen, otherwise use the edit screen
if treatment.past_treatment or upper(service) <> "GET_MEDICATION" then
	Openwithparm(service_window, this, "w_svc_drug_treatment_edit")
else
	Openwithparm(service_window, this, "w_drug_treatment")
end if
```

The w_drug_treatment window is the one I showed above, with the formulation at the top of the page. w_svc_drug_treatment_edit is the one you are seeing now. The difference is that w_svc_drug_treatment_edit is intended for modifying existing prescriptions. Do you want the same window used for both cases, just preloading the information from the existing prescription if there is any?

**Comment by christillman** [comment]
I evaluated the differences between w_svc_drug_treatment_edit and w_drug_treatment. It became evident that w_drug_treatment does support both editing and new drugs already, and that it was improved upon (before me) over what was evidently the original w_svc_drug_treatment_edit. I'm not sure why both were left in the application flow, but I have now removed w_svc_drug_treatment_edit from the flow so only the new window will be used.

**Comment by christillman** [comment]
Similarly, when using the Add Meds button, the treatment is flagged as a past treatment. This causes the w_drug_admin_edit window to be opened, as you illustrated above. The other branch of the code, when not marked as a past treatment, opens the new formulation selection window. So I believe this case should also use w_drug_treatment similarly as above (after selecting the formulation).

**Comment by christillman** [comment]
I evaluated the differences between w_drug_admin_edit and w_drug_treatment. As above, w_drug_treatment does support both editing and new drugs already, and it was improved upon (before me) over what was evidently the original w_drug_admin_edit. I suspect the direction of development had been to combine both w_drug_admin_edit and w_svc_drug_treatment_edit into the new and improved w_drug_treatment. I have now removed w_drug_admin_edit from the flow so only the new window will be used.

w_drug_admin_edit did include a Manual Edit button, which is not on w_drug_treatment. This didn't allow editing of the drug, only editing of the treatment description. It might possibly have been for off-label drug use documentation? If you want to add it to w_drug_treatment I can do that.

The code which opened the w_drug_admin_edit window also did two things after finishing with the window, that aren't done elsewhere.
1) It had the user select a begin date, with the prompt "For How Long?"
2) If the "Past Med Prompt Who Ordered" preference is set (default true), it asked "Who Ordered This Medication?" displaying a list of consultants.
Should these two functions still occur after the drug treatment has been specified, if a past treatment?
